packages/js/src/utils/uri.test.ts

```diff
@@ -0,0 +1,57 @@
+import { CallbackUriError, LogtoError } from './errors';
+import { verifyAndParseCodeFromCallbackUri } from './uri';
+
+const code = 'some_code';
+const state = 'some_state';
+const error = 'some_error';
+const errorDescription = 'some_error_description';
+const redirectUri = 'http://localhost:3000/callback';
+
+describe('verifyAndParseCodeFromCallbackUri', () => {
+  test('valid callback uri should return correct code', () => {
+    const callbackUrl = `http://localhost:3000/callback?code=${code}&state=${state}`;
+    expect(verifyAndParseCodeFromCallbackUri(callbackUrl, redirectUri, state)).toEqual(code);
+  });
+
+  test('callback uri, not starting with redirect uri, should throw', () => {
+    const callbackUrl = `http://localhost:3000/callback?code=${code}&state=${state}`;
+    expect(() =>
+      verifyAndParseCodeFromCallbackUri(callbackUrl, 'http://example.com:3000/callback', state)
+    ).toMatchError(new LogtoError('callbackUri.redirectUriMismatched'));
+  });
+
+  test('callback uri, containing error parameter, should throw', () => {
+    const callbackUrl = `http://localhost:3000/callback?code=${code}&state=${state}&error=${error}`;
+    expect(() => verifyAndParseCodeFromCallbackUri(callbackUrl, redirectUri, state)).toMatchError(
+      new CallbackUriError(error)
```

Comment:
Feels like we should unify the use of these two error types: `CallbackUriError` and `LogtoError`
Either use `CallbackUriError` in all cases or eliminate `CallbackUriError` and use `LogtoError` instead.

Reply:
Removed `CallbackUriError` and updated `LogtoError`, PTAL.

Reply:
Just took a look at the kotlin sdk implementations real quick and found we have taking the first approach.
https://github.com/logto-io/kotlin/pull/75/files
A `CallbackUriVerificationException` has been created and used in every test case.

Reply:
LGTM

Reply:
Added `CallbackUriVerificationError`, PTAL.